common/lib/xmodule/xmodule/js/src/javascript_loader.coffee

```diff
@@ -66,5 +66,52 @@ class @JavascriptLoader
         # just call the completion callback directly, without reloading the file
         completionHandlerGenerator(index)()
       $(placeholder).remove()
-        
-      
+
+  # loadXBlockResources:
+  #   With the provided XBlock resources:
+  #     1) Explicitly attach the resource to the <head> of document
+  #     2) Explicitly wait for each resource to be loaded
+  #     3) Return to callback function when all resources loaded
+  @loadXBlockResources: (resources, callback=null) ->
```

Comment:
I wasn't aware I was duplicating logic, but I agree it would be best to reuse that particular snippet at least, yes.  I'll look into it.